Analysis/Tasks/eventSelection.cxx

```diff
@@ -123,12 +133,97 @@ struct EventSelectionTask {
     }
 
     // Fill event selection columns
-    evsel(alias, bbT0A, bbT0C, bbV0A, bbV0C, bgV0A, bgV0C, bbZNA, bbZNC, bbFDA, bbFDC, bgFDA, bgFDC);
+    int64_t foundFT0 = -1; // this column is not used in run2 analysis
+    evsel(alias, bbT0A, bbT0C, bbV0A, bbV0C, bgV0A, bgV0C, bbZNA, bbZNC, bbFDA, bbFDC, bgFDA, bgFDC, foundFT0);
   }
 };
 
-WorkflowSpec defineDataProcessing(ConfigContext const&)
+// helper table and task for BCs<-->FT0s matching
+namespace o2::aod
 {
-  return WorkflowSpec{
-    adaptAnalysisTask<EventSelectionTask>("event-selection")};
+DECLARE_SOA_INDEX_TABLE_USER(SparseBCsFT0s, BCs, "MA_SP_BCS_FT0S", o2::aod::indices::BCId, o2::aod::indices::FT0Id);
 }
+
+struct EventSelectionTaskRun3Helper {
+  Builds<aod::SparseBCsFT0s> matchedBCsFT0s;
+  void process(aod::BCs, aod::FT0s)
+  {
+  }
+};
+
+struct EventSelectionTaskRun3 {
+  Produces<aod::EvSels> evsel;
+
+  EvSelParameters par;
+
+  void process(aod::Collision const& collision, aod::SparseBCsFT0s const& matchedBCsFT0s, aod::BCs const& bcs, aod::Zdcs const& zdcs, aod::FV0As const& fv0as, aod::FT0s const& ft0s, aod::FDDs const& fdds)
+  {
+    // calculating [minBC, maxBC] range for a collision
+    int deltaBC = std::ceil(collision.collisionTimeRes() / o2::constants::lhc::LHCBunchSpacingNS * 4);
+    int64_t colBC = collision.bc().globalBC();
+    int64_t maxColBC = colBC + deltaBC;
+    int64_t minColBC = colBC - deltaBC;
+
+    int64_t minFt0Dist = INT32_MAX;
+    int64_t foundFT0 = -1;
+    float timeA = -999.f;
+    float timeC = -999.f;
+
+    // calculating BCID iteration range
+    int64_t minGlobalBC = bcs.begin().globalBC();
+    int64_t maxGlobalBC = bcs.iteratorAt(bcs.size() - 1).globalBC();
+    int64_t startIt = minColBC - minGlobalBC;
```

Comment:
This assumes we have no gaps, right?

Reply:
If we have a filling scheme with unfilled BCs, do you then plan to write them to the table anyway?

Reply:
@jgrosseo Yes, we assume no gaps. I thought the agreement was to store bc table continuously irrespective of the filling scheme. In fact CTP will send trigger info (triggerMask etc) for every bc irrespective whether it is colliding or not. We'll need info on A,C and E-like bcs anyway.

Reply:
We still have to do a size estimate of this. You mean you need A,C, E info in case you have signals in a detector?

Reply:
@jgrosseo I would prefer to keep continuous bc/triggerMask for many reasons. First, it can be used for background evaluation. Second, we might need this continuous triggerMask stream for luminosity monitoring, control of trigger afterpulses etc. Of course we could think about removing entries with triggerMask=0 and no signals in forward detectors but I doubt we gain much in terms of data size.

Reply:
Could you make a test using the simulated data and some expected filling schemes? I have a script which prints the compressed/uncompressed size per branch. Let me know if you need it on mattermost.

Reply:
@jgrosseo As far as I remember, Nazar got this script to estimate compressed/uncompressed size per branch. We can try pp scenario, where we expect ~2000 colliding bcs out of total 3564 bcs with typical mu~2%. In order to make realistic test, we should probably assign some realistic triggerMask (e.g. fill with random numbers) for true collisions and fill others with 0.